src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitStorageDao storageDao;
+
+    public PurchaseOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(String fruitName, Integer quantity) {
+        if (storageDao.getQuantity(fruitName) < quantity) {
+            throw new RuntimeException("Not enough " + fruitName + " in the storage");
+        }
```

Comment:
The code calls `storageDao.getQuantity(fruitName)` twice, which could be inefficient if the method involves a costly operation. Consider storing the result in a variable and reusing it to avoid redundant calls.